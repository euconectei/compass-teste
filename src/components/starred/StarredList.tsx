import { StarredType } from '../../types/StarredType';

type StarredListType = {
  starreds: StarredType[];
};

const StarredList = ({ starreds }: StarredListType) => {
  if (!starreds?.length) {
    return <div className='alert alert-info'>Nenhum starred encontrado.</div>;
  }

  return (
    <ul className='nav flex-column'>
      {starreds.map((starred, i) => (
        <li key={`starred-${i}`} className='nav-item'>
          <a href={starred.html_url} className='nav-link'>
            {starred.name}
          </a>
        </li>
      ))}
    </ul>
  );
};

export default StarredList;
