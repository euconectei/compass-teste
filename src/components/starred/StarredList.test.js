import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { starredsMock } from '../../types/__mocks__/starreds.mock';
import StarredList from './StarredList';

describe('Testing StarredList Component', () => {
  it('shold render the component', () => {
    const component = renderer.create(<StarredList starreds={starredsMock} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with an empty starreds', () => {
    render(<StarredList starreds={[]} />);
    expect(screen.getByText('Nenhum starred encontrado.')).toBeInTheDocument();
  });

  it('should render the component with starreds', () => {
    render(<StarredList starreds={starredsMock} />);
    expect(screen.getByText('awesome-hyper')).toBeInTheDocument();
  });
});
