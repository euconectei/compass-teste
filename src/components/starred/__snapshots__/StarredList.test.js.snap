// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing StarredList Component shold render the component 1`] = `
<ul
  className="nav flex-column"
>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/bnb/awesome-hyper"
    >
      awesome-hyper
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/luizalabs/react-boilerplate"
    >
      react-boilerplate
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/jeffersonRibeiro/react-shopping-cart"
    >
      react-shopping-cart
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/doczjs/docz"
    >
      docz
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/webpro/reveal-md"
    >
      reveal-md
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/zehfernandes/pliim"
    >
      pliim
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/viatsko/awesome-vscode"
    >
      awesome-vscode
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/ravenac95/sudolikeaboss"
    >
      sudolikeaboss
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/filosobella/socorrovovo"
    >
      socorrovovo
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/zenorocha/codecopy"
    >
      codecopy
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/mermaid-js/mermaid"
    >
      mermaid
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/nayafia/lemonade-stand"
    >
      lemonade-stand
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/joshbuchea/HEAD"
    >
      HEAD
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/ericdouglas/be-MEAN-resources"
    >
      be-MEAN-resources
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/humhub/humhub"
    >
      humhub
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/liferay/sennajs.com"
    >
      sennajs.com
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/vhf/resrc"
    >
      resrc
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/tinymce/tinymce"
    >
      tinymce
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/thumbor/thumbor"
    >
      thumbor
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/jamesiarmes/php-ews"
    >
      php-ews
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/tableless/Sublime"
    >
      Sublime
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/adampresley/sublime-js-minify"
    >
      sublime-js-minify
    </a>
  </li>
</ul>
`;
